pkg/cmd/roachtest/tests/connection_latency.go

```diff
@@ -145,3 +151,396 @@ func registerConnectionLatencyTest(r registry.Registry) {
+
+	// Copy the CA certificate to `/usr/local/share/ca-certificates/`
+	// to add it as a trusted CA for the host node.
+	trustedCACertCmd := "sudo cp /etc/ssl/certs/ca.crt /usr/local/share/ca-certificates/"
+	err = c.RunE(ctx, option.WithNodes(nodeListOptions), trustedCACertCmd)
+	require.NoError(t, err)
+
+	// Add the custom CA created before as a trusted CA cert for the host node.
+	updateCACertCmd := "sudo update-ca-certificates"
+	err = c.RunE(ctx, option.WithNodes(nodeListOptions), updateCACertCmd)
+	require.NoError(t, err)
+
+	// Restart the slapd service running in the background
+	// to apply all the configurations changed above.
+	restartSlapdCmd := "sudo systemctl restart slapd"
+	err = c.RunE(ctx, option.WithNodes(nodeListOptions), restartSlapdCmd)
+	require.NoError(t, err)
+
+	// Test the ldap connection over TLS
+	cmd := "sudo ldapsearch -H ldaps://localhost -x -b \"\" -s base -LLL -d 1"
+	_, err = c.RunWithDetailsSingleNode(ctx, t.L(),
+		option.WithNodes(c.Node(1)), cmd)
+	require.NoError(t, err)
+}
+
+// Create the directory structure for openLDAP and test LDAP bind
+// using the `ldapwhoami` command.
+func importUserAndGroupsToLDAPServer(
+	ctx context.Context, t test.Test, c cluster.Cluster, nodeListOptions option.NodeListOption,
+) {
+	// Copy the base structure file onto the node
+	content := getTestDataFileContent(
+		t, "./pkg/cmd/roachtest/testdata/ldap_base_structure.ldif")
+	err := c.PutString(ctx, content, "base_structure.ldif", 0755)
+	require.NoError(t, err)
+
+	// Apply the base structure onto the running LDAP service.
+	baseStructureAddCmd := fmt.Sprintf("ldapadd -x -D \"cn=admin,dc=example,dc=com\" -w %s -f base_structure.ldif", ldapAdminPassword)
+	err = c.RunE(ctx, option.WithNodes(nodeListOptions), baseStructureAddCmd)
+	require.NoError(t, err)
+
+	// Copy the user and groups file onto the node.
+	content = getTestDataFileContent(
+		t, "./pkg/cmd/roachtest/testdata/ldap_user_group.ldif")
+	err = c.PutString(ctx, content, "ldap_user_group.ldif", 0755)
+	require.NoError(t, err)
+
+	// Apply the user and group creation onto the running LDAP service.
+	structureAddCmd := fmt.Sprintf("ldapadd -x -D \"cn=admin,dc=example,dc=com\" -w %s -f ldap_user_group.ldif", ldapAdminPassword)
+	err = c.RunE(ctx, option.WithNodes(nodeListOptions), structureAddCmd)
+	require.NoError(t, err)
+
+	testImportCmd := "ldapsearch -x -LLL -b \"dc=example,dc=com\" \"(objectClass=*)\""
+	err = c.RunE(ctx, option.WithNodes(nodeListOptions), testImportCmd)
+	require.NoError(t, err)
+
+	testBindCmd := fmt.Sprintf("ldapwhoami -x -H ldaps://localhost -D \"cn=John Doe,ou=Users,dc=example,dc=com\" -w %s", ldapAdminPassword)
+	err = c.RunE(ctx, option.WithNodes(nodeListOptions), testBindCmd)
+	require.NoError(t, err)
+}
+
+// Setup CRDB specific configs for LDAP authentication.
+// Update the relevant cluster settings for configuring and debugging.
+func setupCRDBForLDAPAuth(ctx context.Context, t test.Test, c cluster.Cluster, hostName string) {
```

Comment:
nit: for all the `SET cluster setting` queries we could introduce a function
```
func setClusterSetting(ctx context.Context, t test.Test, conn *pgx.Conn, setting, value string) {
	_, err := conn.ExecContext(ctx, fmt.Sprintf("SET CLUSTER SETTING %s='%s'", setting, value))
	require.NoError(t, err)
}
```

Reply:
There could also be a separate function to run a given SQL query. The `CREATE ROLE` and `GRANT` statements could go through this.
The `SET CLUSTER SETTING` statements could also go through this function.
<!-- Reviewable comment -OEI9Ysk0DqWZtlZkrv4:bkdwatw -->
<!-- Sent from Reviewable.io -->